libs/search/src/lib/catalog-title/catalog-title.component.ts

```diff
@@ -0,0 +1,20 @@
+import {ChangeDetectionStrategy, Component, OnInit} from '@angular/core'
+
+import {SettingsListResponseApiModel, SiteApiService} from '@lib/gn-api'
+import {Observable} from 'rxjs'
+
+@Component({
+  selector: 'catalog-title',
+  templateUrl: './catalog-title.component.html',
+  styleUrls: ['./catalog-title.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CatalogTitleComponent implements OnInit {
+  siteInfo$: Observable<SettingsListResponseApiModel>
+
+  constructor(private siteApiService: SiteApiService) {}
+
+  ngOnInit(): void {
+    this.siteInfo$ = this.siteApiService.get4()
```

Comment:
Should be shared through a `shareReplay observable` in a dedicated service.
Otherwise, all components that will need siteInfo will perform a request.

Reply:
Ok. There is no cache mechanism like angularjs to not replay http calls on same URLs ?

Reply:
Ah my friend, welcome to the wonderful world of observables... In this case the HTTP request is _only sent when someone subscribes to_ `siteInfo$`. If there are three subscriptions made, the HTTP request is sent three times. There is no built-in caching mechanism that I'm aware of, but you can use an operator for the same result:
```js
 this.siteInfo$ = this.siteApiService.get4().pipe(
  shareReplay()
)
```

See https://blog.strongbrew.io/rxjs-best-practices-in-angular/#sharing-subscriptions for example

Reply:
Done in https://github.com/geonetwork/geonetwork-ui/pull/29/files#diff-c59ec118eac6b193b7c3913e9d915874cda3023f17c08d8845d45c2bf1d11efcR17 but needs to move that to a service - any recommendation for this (location, naming, ...)?

Reply:
Well i don't know yet, we will need a loads of shared observables i guess like the settings or /me or info ...
You could have one or several services for that in common ..
we could create a new lib but i'm not sure it's usefull yet